src/directives/BetaOnboard.tsx

```diff
@@ -103,108 +98,127 @@ const BetaOnboard = ({ directive, mutate }: DirectiveProps) => {
                         void mutate();
                     },
                     async (payload: any) => {
+                        const tenantAlreadyTaken = Boolean(
+                            payload?.job_status?.error?.includes(nameTaken)
+                        );
+
+                        // Handle tracking right away
+                        fireGtmEvent('RegisterFailed', {
+                            tenantAlreadyTaken,
+                            tenant: requestedTenant,
+                        });
                         trackEvent(`${directiveName}:Error`, directive);
+
+                        // Update local state
                         setSaving(false);
-                        setServerError(payload.job_status.error);
+                        setServerError(payload?.job_status?.error);
+                        setNameAlreadyTaken(tenantAlreadyTaken);
                     }
                 );
             }
         },
     };
 
-    const nameAlreadyUsed = useMemo(
-        () => serverError?.includes(nameTaken),
-        [serverError]
-    );
-
     useMount(() => {
         trackEvent(`${directiveName}:Viewed`);
     });
     useUnmount(() => resetOnboardingState());
 
     return (
-        <Stack direction="row">
-            <CustomerQuote hideQuote={belowMd} />
+        <>
+            <Stack
+                spacing={3}
+                sx={{
+                    mt: 1,
+                    mb: 2,
+                    display: 'flex',
+                    alignItems: 'left',
+                }}
+            >
+                <RegistrationProgress
+                    step={2}
+                    loading={saving}
+                    status={status}
+                />
+
+                <HeaderMessage isRegister />
+
+                {serverError ? (
+                    <Box>
+                        <AlertBox
+                            severity="error"
+                            short
+                            title={intl.formatMessage({
+                                id: 'common.fail',
+                            })}
+                        >
+                            {serverError}
+                        </AlertBox>
+                    </Box>
+                ) : null}
+
+                {nameMissing ? (
+                    <Box>
+                        <AlertBox
+                            short
+                            severity="error"
+                            title={<FormattedMessage id="error.title" />}
+                        >
+                            <Typography>
+                                {intl.formatMessage({
+                                    id: 'tenant.errorMessage.empty',
+                                })}
+                            </Typography>
+                        </AlertBox>
+                    </Box>
+                ) : null}
```

Comment:
I flipped the order of the errors around. It was kinda weird to have the error that is 100% directly related to the input further away. So moved server error to the top.